docs/data/system/getting-started/the-sx-prop/the-sx-prop.md

```diff
@@ -292,6 +292,12 @@ If you want to receive the `sx` prop from a custom component and pass it down to
 
 {{"demo": "PassingSxProp.js", "bg": true, "defaultCodeOpen": true}}
 
+## Dynamic values
+
+For highly dynamic CSS values, we recommend using inline CSS variables instead of passing an object with varying values to the `sx` prop on each render. This approach avoids inserting unnecessary `style` tags into the DOM, preventing potential performance issues when dealing with CSS properties that can hold a wide range of values that change frequently. For example, a color picker with live preview.
```

Comment:
We should also talk about the one specific downside here, which is the content security policy to allow inline styles.

Reply:
I wonder if it's worth it. While mentioning CSP can be handy for some readers:
- Is it common enough to mention it?
- Where should we stop talking about these common web dev issues? Following this train of thought, should we mention CORS when talking about loading fonts?

Reply:
I mentioned it because we already have a [page](https://mui.com/material-ui/guides/content-security-policy/) for it. Maybe we can just link to it here at the minimum with a sentence.

Reply:
I see, thanks! I'll add it then

Reply:
Added https://github.com/mui/material-ui/pull/42239/commits/48a251eb8c22d84e969cde6a739772bae639271b